src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -570,16 +570,9 @@ internal JsonClaimSet CreateClaimSet(ReadOnlySpan<char> strSpan, int startIndex,
         {
             int outputSize = Base64UrlEncoding.ValidateAndGetOutputSize(strSpan, startIndex, length);
 
-            byte[] output = ArrayPool<byte>.Shared.Rent(outputSize);
```

Comment:
Ran some benchmarks to compare using and not using ArrayPool for buffers when ⁠[reading header and payload in a token](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/a4f1728593c0393e84d897bc162e5285cc2f484c/src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs#L573-L575). The benchmarks use token with just [⁠a few claims](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/67da84e1d1199e1291716cfc98a7bd6cb6a3b68c/benchmark/Microsoft.IdentityModel.Benchmarks/BenchmarkUtils.cs#L44), so the different could be bigger with larger tokens.
Not using ArrayPool is slightly faster (although insignificant when comparing E2E) and has more allocations and Gen0. Intersting that JwtSecurityTokenHandler_ValidateTokenAsync doesn't show allocation increase.


|Use ArrayPool| Method             | Mean     | Error     | StdDev    |  P90      | P95      | P100     | Gen0   | Allocated |
|--|------------------- |---------:|----------:|----------:|---------:|---------:|---------:|-------:|----------:|
|Yes| ReadJWS_FromMemory | 3.061 μs | 0.0630 μs | 0.1342 μs |  3.293 μs | 3.306 μs | 3.317 μs | 0.1602 |   2.67 KB |
|No| ReadJWS_FromMemory | 2.990 μs | 0.0220 μs | 0.0478 μs | 3.052 μs | 3.059 μs | 3.083 μs | 0.1793 |   2.97 KB |
|Yes| ReadJWS_FromString | 3.036 μs | 0.0095 μs | 0.0205 μs |  3.066 μs | 3.071 μs | 3.073 μs | 0.1602 |   2.67 KB |
|No| ReadJWS_FromString | 2.969 μs | 0.0198 μs | 0.0426 μs | 3.031 μs | 3.035 μs | 3.043 μs | 0.1793 |   2.97 KB |
|Yes| JsonWebTokenHandler_ValidateTokenAsync     | 25.64 μs | 0.024 μs | 0.053 μs | 25.71 μs | 25.72 μs | 25.81 μs | 0.2441 |   4.01 KB |
|No| JsonWebTokenHandler_ValidateTokenAsync     | 25.55 μs | 0.066 μs | 0.143 μs | 25.69 μs | 25.72 μs | 25.78 μs | 0.2441 |    4.3 KB |
|Yes| JwtSecurityTokenHandler_ValidateTokenAsync | 28.46 μs | 0.051 μs | 0.111 μs | 28.60 μs | 28.65 μs | 28.76 μs | 0.7324 |  12.02 KB |
|No| JwtSecurityTokenHandler_ValidateTokenAsync | 28.32 μs | 0.050 μs | 0.109 μs | 28.47 μs | 28.52 μs | 28.55 μs | 0.7324 |  12.02 KB |